curriculum/challenges/english/25-front-end-development/lecture-working-with-the-dom-click-events-and-web-apis/673368ccf52205329b729378.md

```diff
@@ -10,6 +10,140 @@ dashedName: what-is-the-difference-between-innertext-textcontent-and-innerhtml
 
 Watch the lecture video and answer the questions below.
 
+# --transcript--
+
+What is the difference between `innerText`, `textContent`, and `innerHTML`?
+
+Let's learn about `innerText`, `textContent`, and `innerHTML`.
+
+These are properties that you can access in JavaScript to get or change the content of an HTML element.
+
+Even if they may look very similar at first, they do have key differences. Choosing the right one depends on your specific use case, so let's dive in.
+
+Let's start with `innerText`.
+
+`innerText` represents the visible text content of the HTML element and its descendants. 
+
+This property doesn't include hidden text or HTML tags, only rendered text. 
+
+For example, here you can see a `div` element that contains two paragraphs.
+
+```html
+<div id="container">
+  <p>Hello, World!</p>
+  <p>I'm learning JavaScript</p>
+</div>
+```
+
+If we get a reference to this HTML element in our JavaScript code using `getElementById()`, we can access the `innerText` property of this element.
+
+```js
+const container = document.getElementById("container");
+console.log(container.innerText);
+```
+
+This is the inner text of this element.
+
+```md
+Hello, World!
+I'm learning JavaScript
+```
+
+The property returns a string with the text contained within the element, including text from its descendants.
+
+You should know that `innerText` only returns the text that is visible at the particular moment when the string is requested. If a child element is hidden, its text won't be visible.
+
+This is an example where the second paragraph is hidden.
+
+```html
+<div id="container">
+  <p>Hello, World!</p>
+  <p hidden>I'm learning JavaScript</p>
+</div>
+```
+
+If we try to log the `innerText` again, now the output won’t have the text of the second paragraph.
+
+```js
+console.log(container.innerText);
+```
+
+This will be the output:
+
+```md
+Hello, World!
+```
+
+You can set the `innerText` of an HTML element like this, but this will replace the existing text and add a line break element (`br`) element for every line break.
```

Comment:
```suggestion
You can set the `innerText` of an HTML element like this, but this will replace the existing text and add a line break element (`br`) element for every line break:
```